### PR TITLE
Remove dependency on pytest-subtests.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR changes a small number of tests that relied on the `pytest-subtests` plugin.

## Motivation and Context
The `pytest-subtests` plugin was causing some trouble because upgrades to `pytest` broke compatibility with `pytest-subtests` on two separate occasions:
1. 0.3.1 to 0.3.2 breakage is documented here: #366, https://github.com/glotzerlab/signac/pull/366#issuecomment-676579490.
2. 0.3.0 to 0.3.1 breakage is not documented but I remember running into it on my own, and found a comment to corroborate this on Slack.

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change<sup>1</sup>

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt) and added all related issue and pull request numbers for future reference (if applicable). See example below.

### CODE REVIEW

File: tests/test_project.py [hunk 1]
```diff
@@ -1480,85 +1480,68 @@ def test_import_own_project(self):
...
-    def test_repr_many_jobs_heterogeneous(self, subtests):
-        # Many jobs with many different state points
-        for i in range(self.num_many_jobs):
-            for v in self.valid_sp_values:
-                self.project.open_job(dict(a=v)).init()
-        self.call_repr_methods(subtests)
+    @pytest.mark.parametrize("project_generator,num_jobs", project_repr_generators)
+    def test_project_repr_methods(self, project_generator, num_jobs):
+        project_generator(self.project, num_jobs)
+        str(self.project)
```
Comment (csadorf) [inline tests/test_project.py:1517]: Should we do a super simple sanity check here, something like `assert len(self.project) > 0` or `assert 'project' in str(self.project)` ?

File: tests/test_project.py [hunk 2]
```diff
@@ -1480,85 +1480,68 @@ def test_import_own_project(self):
...
+                if use_pandas and not PANDAS:
+                    raise pytest.skip('requires use_pandas')
+                self.project.find_jobs(filter_)._repr_html_()
+
+    @pytest.mark.parametrize("project_generator,num_jobs", project_repr_generators)
+    def test_Schema_repr_methods(self, project_generator, num_jobs):
+        project_generator(self.project, num_jobs)
+        schema = self.project.detect_schema()
+        str(schema)
+        repr(schema)
```
Comment (csadorf) [inline tests/test_project.py:1543]: Similar to my comment above.

**Review by csadorf — APPROVED** [review]

File: tests/test_project.py (hunk 1 above)
Comment (bdice) [reply tests/test_project.py]: Good suggestion! I added `assert len(str(self.project)) > 0` and some related tests. The other case is already handled twice by `TestProject.test_get_id` and `TestProject.test_property_id`.

**Comment by bdice** [comment]
@klywang Can you please review?

File: tests/test_project.py [hunk 3]
```diff
@@ -1480,85 +1480,71 @@ def test_import_own_project(self):
             assert len(tmp_project) == len(self.project)
 
 
-class TestProjectRepresentation(TestProjectBase):
+def add_jobs_homogeneous(project, num_jobs):
+    # Add jobs with many different state points
+    valid_sp_values = [None, 0, 1, 0.0, 1.0, True, False, [0, 1, 2], [0, 1.0, False]]
```
Comment (klywang) [inline tests/test_project.py:1485]: If the goal is to always test the same values for `valid_sp_values` for `add_jobs_homogeneous` and `add_jobs_heterogeneous`, would it make sense to define `valid_sp_values` outside of both functions and take them in as inputs?

**Review by klywang — APPROVED** [review]
Everything looks good!

File: tests/test_project.py (hunk 3 above)
Comment (bdice) [reply tests/test_project.py]: Good idea! I'll define a module-level constant.

**Pull request merged**